### PR TITLE
Issues/#5925 certificate avoid code duplication

### PR DESCRIPTION
**Pull request by farmborough**
Fixes https://github.com/gravitee-io/issues/issues/5925.

### CODE REVIEW

File: gravitee-am-certificate/gravitee-am-certificate-javakeystore/src/main/java/io/gravitee/am/certificate/javakeystore/provider/JavaKeyStoreProvider.java [hunk 1]
```diff
@@ -15,258 +15,68 @@
...
-    private SignatureAlgorithm signature = SignatureAlgorithm.RS256;
-    private io.gravitee.am.certificate.api.Key certificateKey;
-    private List<CertificateKey> certificateKeys;
-
+public class JavaKeyStoreProvider extends AbstractCertificateProvider implements InitializingBean {
     @Autowired
     private JavaKeyStoreConfiguration configuration;
-
     @Autowired
     private CertificateMetadata certificateMetadata;
```
Comment (leleueri) [inline gravitee-am-certificate/gravitee-am-certificate-javakeystore/src/main/java/io/gravitee/am/certificate/javakeystore/provider/JavaKeyStoreProvider.java:36]: Is it possible to considere this attribute to be part of the Abstract class as it is also used into the PKCS12Provider ?

**Review by leleueri — CHANGES_REQUESTED** [review]

File: gravitee-am-certificate/gravitee-am-certificate-javakeystore/src/main/java/io/gravitee/am/certificate/javakeystore/provider/JavaKeyStoreProvider.java [hunk 2]
```diff
@@ -15,258 +15,68 @@
...
-                        .get());
+    public CertificateMetadata certificateMetadata() {
+        return certificateMetadata;
     }
 
     @Override
-    public Single<List<CertificateKey>> publicKeys() {
-        return Single.just(certificateKeys);
+    protected String getStorepass() {
+        return configuration.getStorepass();
```
[the remark this answers is not in the document]
Comment (farmborough) [reply gravitee-am-certificate/gravitee-am-certificate-javakeystore/src/main/java/io/gravitee/am/certificate/javakeystore/provider/JavaKeyStoreProvider.java:60]: I tried that first but ran this caused an error when the certificate json was deserialised. The problem is that the java certificate schema and PKCS12 schemas are identical, apart from the key property itself (it's `content` for PKCS12, 'jks` for Java). The way around that was to change the schemas as well, although that then hints that the two certificate implementations are redundant.

File: gravitee-am-certificate/gravitee-am-certificate-javakeystore/src/main/java/io/gravitee/am/certificate/javakeystore/provider/JavaKeyStoreProvider.java (hunk 1 above)
Comment (farmborough) [reply gravitee-am-certificate/gravitee-am-certificate-javakeystore/src/main/java/io/gravitee/am/certificate/javakeystore/provider/JavaKeyStoreProvider.java:36]: I can do that, so long as adding the Spring dependency to that module isn't a problem?

File: gravitee-am-certificate/gravitee-am-certificate-javakeystore/src/main/java/io/gravitee/am/certificate/javakeystore/provider/JavaKeyStoreProvider.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (leleueri) [reply gravitee-am-certificate/gravitee-am-certificate-javakeystore/src/main/java/io/gravitee/am/certificate/javakeystore/provider/JavaKeyStoreProvider.java:58]: ok thanks you for the details

File: gravitee-am-certificate/gravitee-am-certificate-javakeystore/src/main/java/io/gravitee/am/certificate/javakeystore/provider/JavaKeyStoreProvider.java (hunk 1 above)
Comment (leleueri) [reply gravitee-am-certificate/gravitee-am-certificate-javakeystore/src/main/java/io/gravitee/am/certificate/javakeystore/provider/JavaKeyStoreProvider.java]: As Spring is used on REST API & GW, I don't think it is a problem IMHO.

**Review by leleueri — APPROVED** [review]

**Pull request merged**